File: src/main/java/org/jukeboxmc/block/behavior/BlockStainedGlass.java
package org.jukeboxmc.block.behavior;

import com.nukkitx.nbt.NbtMap;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.block.Block;
import org.jukeboxmc.block.data.BlockColor;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.item.behavior.ItemStainedGlass;
import org.jukeboxmc.util.Identifier;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockStainedGlass extends Block {

    public BlockStainedGlass( Identifier identifier ) {
        super( identifier );
    }

    public BlockStainedGlass( Identifier identifier, NbtMap blockStates ) {
        super( identifier, blockStates );
    }

    @Override
    public Item toItem() {
        return Item.<ItemStainedGlass>create( ItemType.STAINED_GLASS ).setColor( this.getColor() );
    }

    public @NotNull BlockStainedGlass setColor(@NotNull BlockColor color ) {
        this.setState( "color", color.name().toLowerCase() );
        return this;
    }

    public @NotNull BlockColor getColor() {
        return this.stateExists( "color" ) ? BlockColor.valueOf( this.getStringState( "color" ) ) : BlockColor.WHITE;
    }
}

File: src/main/java/org/jukeboxmc/block/behavior/BlockStainedGlass.kt
package org.jukeboxmc.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.block.Block
import org.jukeboxmc.block.data.BlockColor
import org.jukeboxmc.item.Item
import org.jukeboxmc.item.ItemType
import org.jukeboxmc.item.behavior.ItemStainedGlass
import org.jukeboxmc.util.Identifier
import java.util.Locale

/**
 * @author LucGamesYT
 * @version 1.0
 */
class BlockStainedGlass : Block {
    constructor(identifier: Identifier) : super(identifier)
    constructor(identifier: Identifier, blockStates: NbtMap?) : super(identifier, blockStates)

    override fun toItem(): Item {
        return Item.create<ItemStainedGlass>(ItemType.STAINED_GLASS).setColor(
            color,
        )
    }

    fun setColor(color: BlockColor): BlockStainedGlass {
        setState<Block>("color", color.name.lowercase(Locale.getDefault()))
        return this
    }

    val color: BlockColor
        get() = if (stateExists("color")) BlockColor.valueOf(getStringState("color")) else BlockColor.WHITE
}
